Tests/UnitTests/CMakeLists.txt

```diff
@@ -25,6 +25,68 @@ macro(add_unittest _name)
   add_test(NAME ${_name} COMMAND ${_target})
 endmacro()
 
+function(add_non_compile_test name src)
```

Comment:
Could you add a few lines of comment here to describe the high-level approach of these tests?
I think this is a quite complicated bit of `cmake` code, and I think that would help future developers quite a bit.